Objective-C/CBLDocument.mm

```diff
@@ -210,9 +253,9 @@ - (void) replaceC4Doc: (CBLC4Document*)c4doc {
 
 #pragma mark - Fleece Encoding
 
-- (FLSliceResult) encodeWithRevFlags: (C4RevisionFlags*)outRevFlags error:(NSError**)outError {
+- (FLSliceResult) encodeWithRevFlags: (C4RevisionFlags*)outRevFlags useSharedEncoder: (BOOL)sharedEncoder error:(NSError**)outError {
```

Comment:
useSharedEncoder: is not needed as the code can use _isRemoteDoc to check whether the shared encoder should be used or not.